### ISSUE TITLE
update broken and outdated examples

### ISSUE THREAD

**Issue by showell** [jashkenas/coffeescript#1707]
Edit: This thread started with a fairly narrow issue, then the discussion broadened to the state of the "examples" directory.
The proposal here is to fix two small problems in the computer_science directory (one of which Michael just fixed, thank you), and then to deprecate some examples that are probably not worth maintaining going forward.

Original post:

The linked_list.coffee example in examples/computer_sciences uses a bare constructor that is misleading:

The code below doesn't do it what it suggests, and, in fact, you can remove the code and the example still works.

```
->
  this._head = null # Pointer to the first item in the list.
```

**Comment by GeoffreyBooth** [comment]
I’m updating the documentation in preparation for the 2.0.0 release. Anyone watching this thread care to tell me what you’d like to keep or remove from `/examples`? I know, it’s been gnawing at you for five years, now is your chance!

**Comment by GeoffreyBooth** [comment]
Closed in #4367
